lib/ecto/adapters/sql.ex

```diff
@@ -1351,22 +1398,23 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
-  defp log_stacktrace(stacktrace, repo) do
-    with [_ | _] <- stacktrace,
-         {module, function, arity, info} <- last_non_ecto(Enum.reverse(stacktrace), repo, nil) do
+  defp log_stacktrace([_ | _] = stacktrace, repo, {module, function, args}) do
+    entries = apply(module, function, [stacktrace, repo | args])
```

Comment:
```suggestion
    entries = apply(module, function, [stacktrace, %{repo: repo} | args])
```

As per the Ecto issue.